src/formatter.ts

```diff
@@ -64,12 +67,21 @@ function formatScalar(v: Value): string {
   return JSON.stringify(v);
 }
 
-function formatAnnotationValue(a: AttributeValue): string {
-  if (a.name === 'primary') return formatScalar(a.value);
+function formatAnnotationValue(a: AttributeValue): string | undefined {
+  const formattedValue = formatScalar(a.value);
+
+  if (a.name === 'primary') return formattedValue;
   if (a.name === 'id' && typeof a.value === 'string' && isIdentifier(a.value))
     return '#' + a.value;
   if (a.type === 'class' && isIdentifier(a.name)) return '.' + a.name;
-  return `${a.name}=${formatScalar(a.value)}`;
+
+  if (formattedValue !== undefined) {
+    return `${a.name}=${formattedValue}`;
+  }
+  // The Markdoc parser does not support undefined attribute
+  // values. Filter those values out.
+  return undefined;
```

Comment:
Wait, I just realized. Let's just move `if (formattedValue === undefined) return undefined` to the first branch of this function, and leave `return `${a.name}=${formattedValue}`;` as the fallback.